icgcget/cli.py

```diff
@@ -79,123 +95,211 @@ def cli(ctx, config, logfile):
     config_file = config_parse(config)
     if config != DEFAULT_CONFIG_FILE and not config_file:
         raise click.BadParameter(message="Invalid config file")
+
     ctx.default_map = config_file
 
     if logfile is not None:
         logger_setup(logfile)
-    elif config_file.has_key('logfile'):
+    elif 'logfile' in config_file:
         logger_setup(config_file['logfile'])
     else:
         logger_setup(None)
 
 
 @cli.command()
-@click.argument('repo', type=click.Choice(REPOS))
-@click.argument('fileid', nargs=-1)
-@click.option('--manifest', '-m', default=False)
-@click.option('--output', type=click.Path(exists=False))
-@click.option('--cghub-access')
-@click.option('--cghub-path')
-@click.option('--cghub-transport-parallel')
-@click.option('--ega-access')
-@click.option('--ega-password')
-@click.option('--ega-path')
-@click.option('--ega-transport-parallel')
-@click.option('--ega-udt')
-@click.option('--ega-username')
-@click.option('--gdc-access')
-@click.option('--gdc-path')
-@click.option('--gdc-transport-parallel')
-@click.option('--gdc-udt')
-@click.option('--icgc-access')
-@click.option('--icgc-path')
-@click.option('--icgc-transport-file-from')
-@click.option('--icgc-transport-memory')
-@click.option('--icgc-transport-parallel')
-def download(repo, fileid, manifest, output,
+@click.argument('fileids', nargs=-1, required=True)
+@click.option('--repos', '-r', type=click.Choice(REPOS), multiple=True)
+@click.option('--manifest', '-m', is_flag=True, default=False)
+@click.option('--output', type=click.Path(exists=True, writable=True, file_okay=False, resolve_path=True))
+@click.option('--cghub-access', type=click.STRING)
+@click.option('--cghub-path', type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+@click.option('--cghub-transport-parallel', type=click.STRING)
+@click.option('--ega-password', type=click.STRING)
+@click.option('--ega-path', type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+@click.option('--ega-transport-parallel', type=click.STRING)
+@click.option('--ega-udt', type=click.BOOL)
+@click.option('--ega-username', type=click.STRING)
+@click.option('--gdc-access', type=click.STRING)
+@click.option('--gdc-path', type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+@click.option('--gdc-transport-parallel', type=click.STRING)
+@click.option('--gdc-udt', type=click.BOOL)
+@click.option('--icgc-access', type=click.STRING)
+@click.option('--icgc-path', type=click.Path(exists=True, dir_okay=False, resolve_path=True))
+@click.option('--icgc-transport-file-from', type=click.STRING)
+@click.option('--icgc-transport-parallel', type=click.STRING)
+@click.option('--yes-to-all', '-y', is_flag=True, default=False, help="Bypass all confirmation prompts")
```

Comment:
A prompt is something that requires `stdin`. Is that the case here? What is the user saying "yes" to?